documentation/src/main/asciidoc/ch06.asciidoc

```diff
@@ -210,6 +210,66 @@ include::{sourcedir}/org/hibernate/validator/referenceguide/chapter06/MyFutureVa
 You should only implement one of the `initialize()` methods. Be aware that both are called when initializing the validator.
 ====
 
+[[constraint-validator-payload]]
+===== Passing a payload to the constraint validator
+
+From time to time, you might want to condition the constraint validator behavior on some external parameters.
+
+For instance, your zip code validator could vary depending on the locale of your application instance if you have one
+instance per country.
+Another requirement could be to have different behaviors on specific environments: the staging environment may not have
+access to some external production resources necessary for the correct functioning of a validator.
+
+The notion of constraint validator payload was introduced for all these use cases.
+It is an object passed from the `Validator` instance to each constraint validator via the `HibernateConstraintValidatorInitializationContext`.
+
+The example below shows how to set a constraint validator payload during the `ValidatorFactory` initialization.
+Unless you override this default value, all the ``Validator``s created by this `ValidatorFactory` will have this
+constraint validator payload value set.
+
+[[example-constraint-validator-payload-definition-validatorfactory]]
+.Defining a constraint validator payload during the `ValidatorFactory` initialization
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/validator/referenceguide/chapter06/constraintvalidatorpayload/ConstraintValidatorPayloadTest.java[tags=setConstraintValidatorPayloadDuringValidatorFactoryInitialization]
+----
+====
+
+Another option is to set the constraint validator payload per `Validator` using a context:
+
+[[example-constraint-validator-payload-definition-validatorcontext]]
+.Defining a constraint validator payload using a `Validator` context
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/validator/referenceguide/chapter06/constraintvalidatorpayload/ConstraintValidatorPayloadTest.java[tags=setConstraintValidatorPayloadInValidatorContext]
+----
+====
+
+Once you have set the constraint validator payload, it is time to use it in your constrait validators as shown in the example below:
+
+[[example-constraint-validator-payload-usage]]
+.Using the constraint validator payload in a constraint validator
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/validator/referenceguide/chapter06/constraintvalidatorpayload/ZipCodeValidator.java[tags=include]
+----
+====
+
+`HibernateConstraintValidatorInitializationContext#getConstraintValidatorPayload()` has a type parameter
```

Comment:
Question on that one, what happens if I have registered two payloads of type `A` and `B`, `B` extends `A` and I request `A`? Is an order of precedence defined?

Reply:
If I understand your question right you can't have two payloads, because the last one wins.